### ISSUE TITLE
Error in MEDIUM_NoteBook/NeuralNetSeq_FeatureImportance/NeuralNetSeq_FeatureImportance.ipynb

### ISSUE THREAD

**Issue by korosig** [cerlymarco/MEDIUM_NoteBook#20]
HI there,
I like your GitHub repo!

I want to implement your RNN-Heatmap visualization in my task. During the testing I have found, you are using the wrong indices in the **activation_grad** function.  In your example, the OUTPUT shape is 15,32 while the dimension of the weight is 15. So when the  **activation_grad(seq, model)** function create the **cam** variable use first 15 features from 32.  While somebody (like me) uses output with shape 500,20 it will create an error, because in that scenario the dimension of the weight will be 500. 

I attached a pic of the problem.

![image](https://user-images.githubusercontent.com/23563516/101788017-fd8ed580-3aff-11eb-9075-8d816f8d8cc8.png)

**Comment by korosig** [comment]
And as I understand 

this doesn't change the value of the **cam** variable


```
    # Get a ponderated map of filters according to grad importance
    cam = np.ones(output.shape[0], dtype=np.float32)
    for index, w in enumerate(weights):
        cam += w * output[:, index]
```

**Comment by cerlymarco** [comment]
Hi, 

15 are not the features. In the example is the temporal dimensionality. We are operating the importance map on the temporal dimensionality and not on the dimensionality of the features.

However, what's the error that is generated?

**Comment by cerlymarco** [comment]
Hi, thanks for reporting

You need to change:

weights = tf.reduce_mean(grads, axis=1)  # (15,)

into

weights = tf.reduce_mean(grads, axis=0)   # (32,)

I also uploaded a new version of the notebook

**Comment by korosig** [comment]
> Hi, thanks for reporting
> 
> You need to change:
> 
> weights = tf.reduce_mean(grads, axis=1) # (15,)
> 
> into
> 
> weights = tf.reduce_mean(grads, axis=0) # (32,)
> 
> I also uploaded a new version of the notebook

And what if the output.shape[1] is less than length of weights?  In my scenario that is what happend.

**Comment by korosig** [comment]
Thank you for your help ..... it is the most useable GitRepo I have ever seen :-)
